exec/java-exec/src/main/java/org/apache/drill/exec/client/DrillClient.java

```diff
@@ -411,6 +420,98 @@ public void runQuery(QueryType type, List<PlanFragment> planFragments, UserResul
   }
 
   /**
+   * Get the list of catalogs in <code>INFORMATION_SCHEMA.CATALOGS</code> table satisfying the given filters.
+   *
+   * @param catalogNameFilter Filter on <code>catalog name</code>. Pass null to apply no filter.
+   * @return
+   */
+  public DrillRpcFuture<GetCatalogsResp> getCatalogs(LikeFilter catalogNameFilter) {
```

Comment:
These APIs are supposed to be fast, not sure implementing cancel provides any value.